ctapipe/image/toymodel.py

```diff
@@ -79,11 +78,16 @@ def obtain_time_image(x, y, centroid_x, centroid_y, psi, time_gradient, time_int
         Pulse time in nanoseconds at (x, y)
 
     """
-    longitudinal, _ = camera_to_shower_coordinates(x, y, centroid_x, centroid_y, psi)
-    longitudinal_m = longitudinal.to_value(u.m)
-    time_gradient_ns_m = time_gradient.to_value(u.ns / u.m)
-    time_intercept_ns = time_intercept.to_value(u.ns)
-    return longitudinal_m * time_gradient_ns_m + time_intercept_ns
+    unit = x.unit
```

Comment:
I think I prefer the old code structure (without the no_units function). It is easier to understand imo without the overhead of the function call.